### PR TITLE
This fixes a crash while typing a positional pattern.

### LINKED ISSUE #26467: Assertion hit in  MakeTestsAndBindings
```C#
                    Debug.Assert(recursive.HasAnyErrors); // line 451
```

image

I think the assertion is reached before I typed the last character, so it would be `_ = a is ()`.

image

Tagging @gafter

### PR DESCRIPTION
**Pull request by gafter**
This fixes a crash found while typing in the IDE. It is a tiny bug fix.
Fixes #26467

@jcouv @agocke @cston @dotnet/roslyn-compiler May I please have a couple of reviews?

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs [hunk 1]
```diff
@@ -554,6 +554,11 @@ private BoundPattern BindDeconstructionPattern(DeconstructionPatternSyntax node,
                 BoundExpression deconstruct = MakeDeconstructInvocationExpression(
                     node.SubPatterns.Count, inputPlaceholder, node, diagnostics, outPlaceholders: out ImmutableArray<BoundDeconstructValuePlaceholder> outPlaceholders);
                 deconstructMethod = deconstruct.ExpressionSymbol as MethodSymbol;
+                if (deconstructMethod == null)
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs]: `deconstructMethod` is a `MethodSymbol`. Use `is null` or `(object)`.

Comment (agocke) [reply src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs]: Below you check `outPlaceholders.IsDefaultOrEmpty` to see if the deconstruction is an error, while here you check if the method is `null`. It seems like this code could be simplified significantly by settling on a single check and reusing it, including where we check `isError` below.

Comment (gafter) [reply src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs]: That is a test for a particular subelement of the deconstruction being an error, not the whole thing. Checking that the method is null is one way the whole thing can be in error.

---
In reply to: [184870737](https://github.com/dotnet/roslyn/pull/26481#discussion_r184870737) [](ancestors = 184870737)

**Comment by gafter** [comment]
Responded to review comments.

@agocke @jcouv May I please have a second review?

**Pull request merged**